cli/src/commands/parse.rs

```diff
@@ -396,48 +316,120 @@ fn relative_path(from: &Path, to: &Path) -> Result<PathBuf> {
     Ok(result)
 }
 
+/// Construct command for sandboxed lockfile parsing.
+fn parse_sandboxed_command(
+    path: &Path,
+    display_path: &str,
+    lockfile_type: Option<&String>,
+    generate_lockfile: bool,
+    skip_sandbox: bool,
+) -> Result<Command> {
+    let current_exe = env::current_exe()?;
+    let mut command = Command::new(current_exe);
+
+    command.arg("parse-sandboxed");
+    command.arg(path);
+    command.arg(display_path);
+
+    if let Some(lockfile_type) = lockfile_type {
+        command.args(["--type", &lockfile_type.to_string()]);
+    }
+
+    if generate_lockfile {
+        command.arg("--generate-lockfile");
+    }
+
+    if skip_sandbox {
+        command.arg("--skip-sandbox");
+    }
+
+    Ok(command)
+}
+
+/// Create sandbox for dependency file parsing.
+///
+/// This sandbox will automatically add all exceptions necessary to generate
+/// lockfiles for any ecosystem.
+fn depfile_parsing_sandbox(canonical_manifest_path: &Path) -> Result<Birdcage> {
```

Comment:
The parsing sandbox doesn't include access to Phylum's config file which results in a few nasty log messages

```
❗ Config access denied: Operation not permitted (os error 1)
[2023-12-21T16:05:16Z ERROR phylum] Failed to save config: Operation not permitted (os error 1)
```

These aren't usually printed since we capture stderr and they don't cause the process to fail (it uses a default config instead), but I noticed it when I caused a separate error in the subprocess...

This seems a bit complicated to resolve because the config file path can be overridden on the command line. So I suppose we should give access to the effective config file and pass the config file argument to the subprocess...

Or, since the config doesn't actually affect the behavior of `phylum parse-sandboxed`, we could disable config loading for the subprocess (but that would require adding a `--no-config` top-level option).

And finally, because the default config doesn't have a `last_update` timestamp... The subprocess does an update check...

```
       ---------------- Update Available ----------------

       A new version of the Phylum CLI is available. Run

       	phylum update

       to update to the latest version!

       --------------------------------------------------
```

It be nice to avoid that too...

Reply:
I could also remove the error, how likely are people running into this anyway?